applications/ShapeOptimizationApplication/python_scripts/mapper_factory.py

```diff
@@ -0,0 +1,29 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgärtner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# importing the Kratos Library
+from KratosMultiphysics import *
+from KratosMultiphysics.ShapeOptimizationApplication import *
+
+# check that KratosMultiphysics was imported in the main script
+CheckForPreviousImport()
+
+# ==============================================================================
+def CreateMapper( designSurface, optimizationSettings ):
+    isMatrixFreeMappingRequired = optimizationSettings["design_variables"]["filter"]["matrix_free_filtering"].GetBool()
```

Comment:
the choice of "matrix_free_filtering" should not be a property of "filter", in my opinion it belongs one level above into "design_variables". 
Here also future settings like:
"integration_type"
"consistent_mapping"
... could be placed

or one could add a "mapper_settings" to the "design_variables"